arm64 support from prebuilt gcc packages

skarlsson opened the pull request:
- adds multiple toolchains support
- adds arm64 from toolchains_gcc_packages

skarlsson committed:
initial arm64 support from prebuilt toolchain

skarlsson committed:
point to official release

skarlsson committed:
cleanup

skarlsson committed:
cleanup

nradakovic commented on extentions/gcc.bzl:
```diff
@@ -21,49 +21,124 @@ def _gcc_impl(mctx):
-            name = "%s_gcc" % toolchain_info["name"],
+            name = "%s_gcc" % name,
             urls = [toolchain_info["url"]],
-            build_file = "@score_toolchains_gcc//toolchain/third_party:gcc.BUILD",
```
We need to have option to set this file from outside. We cannot know what kind of structure any future GCC can have so it would be bad from us to hardcode paths to the tooling. We can leave it as default option, but not as only option. Also, some of the tools are missing, like GOLD linker for example.

skarlsson replied:
fixed the first part -  option to set this file from outside

nradakovic commented on platforms/BUILD:
```diff
@@ -19,3 +19,12 @@ platform(
     ],
     visibility = ["//visibility:public"],
 )
+
```
We have platform repository which we need to reuse for platform definitions. It's still not released, but that's just matter of adding it to S-CORE Bazel Registry.
Bazel Platform repo: https://github.com/eclipse-score/bazel_platforms

FScholPer replied:
@skarlsson Whats the state of that pull request? That would be really great to have it in. Can you please fix the comments especially the platforms. Release is already there bazel_dep(name = "score_bazel_platforms", version = "0.0.2")

skarlsson replied:
I took the earlier comment as a hint to wait until the you had a new implementation in place and I thought to give it a try again at a later time.

skarlsson replied:
The pull request - at the time it was done worked and I had a working arm64 baselibs on top of that - but I found it inefficient to have so many local forks so I decided to wait

nradakovic commented on rules/gcc.bzl:
```diff
@@ -37,12 +37,23 @@ def _impl(rctx):
         rctx (repository_ctx): The Bazel repository context, providing access to attributes
             and methods for creating repository rules.
     """
+    # Map architecture names to platform CPU values
```
I don't understand this. What is the purpose of it?

skarlsson replied:
fixed

nradakovic reviewed (commented):
The current implementation is soon to be deprecated and due to that we need to move it to:
https://github.com/bazelbuild/rules_cc/tree/main/examples/rule_based_toolchain

skarlsson committed:
Removed redundant arch_map - The mapping was 1:1 and served no purpose
Added build_file override capability - Allows custom BUILD files for non-standard GCC structures
Removed unused template files - Cleaned up the toolchain/third_party directory
